addons/account/wizard/base_document_layout.py

```diff
@@ -1,13 +1,80 @@
+    account_number = fields.Char(compute='_compute_account_number', inverse='_inverse_account_number',)
+
     def document_layout_save(self):
         """Save layout and onboarding step progress, return super() result"""
         res = super(BaseDocumentLayout, self).document_layout_save()
         if step := self.env.ref('account.onboarding_onboarding_step_base_document_layout', raise_if_not_found=False):
             for company_id in self.company_id:
                 step.with_company(company_id).action_set_just_done()
+            # When we finish the configuration of the layout, we want the dialog size to be reset to large
+            # which is the default behaviour.
+            if res.get('context'):
+                res['context']['dialog_size'] = 'large'
         return res
+
+    @api.depends('report_layout_id')
+    def _compute_from_invoice(self):
+        """
+            This will be computed when the configure layout wizard is launched, so we cannot have multiple
+            moves in self.
+        """
+        move = self.env['account.move'].browse(self.env.context.get('active_id'))
+        is_from_move = self.env.context.get('active_model') == 'account.move'
+        self.from_invoice = is_from_move and move.move_type == 'out_invoice'
+
+    @api.depends('qr_code')
+    def _compute_preview(self):
+        """ This override is needed to add the depends on the qr code and the move """
+        super()._compute_preview()
+        if self.env.context.get('active_model') != 'account.move':
+            return
+        move = self.env['account.move'].browse(self.env.context.get('active_id'))
+        # If there is a move in the context then, the value put in the preview come from the move
+        if not move.exists():
+            return
+
+        styles = self._get_asset_style()
+        for wizard in self:
+            if wizard.report_layout_id:
+                preview_css, wizard_with_logo = self._get_render_information(styles, wizard)
+
+                # We don't want to display the qr_code twice since we decided to put a fake one.
+                move.display_qr_code = False
+                self.preview = wizard_with_logo.env['ir.ui.view']._render_template('account.report_invoice_wizard_iframe', {
+                    'company': wizard_with_logo,
+                    'preview_css': preview_css,
+                    'is_html_empty': is_html_empty,
+                    'o': move,
+                    'qr_code': self.qr_code,
+                })
+
+    @api.depends('partner_id', 'account_number')
+    def _compute_account_number(self):
+        for record in self:
+            if record.partner_id.bank_ids:
+                record.account_number = record.partner_id.bank_ids[0].account_number or ''
+            else:
+                record.account_number = ''
+
+    def _inverse_account_number(self):
+        for record in self:
+            if record.qr_code:
```

Comment:
Could be inversed to have something like this:
```suggestion
            if not record.qr_code:
                continue
```
And reduce the indent level of one